python/tests/stress/arcticdb/version_store/test_mem_leaks.py

```diff
@@ -146,13 +154,132 @@ def grow_exp(df_to_grow: pd.DataFrame, num_times_xx2: int):
         df_to_grow = pd.concat([df_to_grow, df_prev])
     return df_to_grow
 
+def generate_big_dataframe(rows:int=1000000):
+    print("Generating big dataframe")
+    st = time.time()
+    df = get_sample_dataframe(rows)
+    df = grow_exp(df, 5)
```

Comment:
Generating milions of lines with get_sample_dataframe() is perhaps slower than generating 10k, then appending same numbur and to result appending the result from previous op. I think that was the reason I tried to speed up things.

Reply:
I will ... btw ... actually in many cases accept that reasoning comes from experimentation. Like in that case You cannot predict how much memory is going to leak thus you have to make something large enough to both catch a problem that we had with massive leakage, and smaller ones, but at the same time keep time for execution reasonable :-)

Thus this code is needed for not memray approach based on repetitions, measurments with psutil (which turn out to be relative because gc.collect() will not do garbage collection, or even if it do it it will be in python code only and have no control on C++ part) Thus those types of tests heuristics based on knowledge and observation of the creator.

But at the same time as memray does not support windows having this approach avail is nice. Even that our Windows runners have less HDD memory than needed for the test to execute. :-(

Reply:
def generate_big_dataframe(rows:int=1000000, num_exp_time_growth:int=5) -> pd.DataFrame:
    """
    A quick and time efficient wat to generate very large dataframe.
    The first parameter will be passed to get_sample_dataframe() so that a dataframe
    of that number of rows is generated. Later this df will be used to grow exponentially (power of 2)
    the number of rows on the final dataframe by concatenating N times the result dataframe to itself
    That process if >6-8 times more time efficient. The down side is that you have N**2 repetitions of
    same original table. But for many usages size is what we need not uniqueness
    """